Chaincase.UI/ViewModels/SendViewModel.cs

```diff
@@ -255,20 +279,99 @@ private void SetAmountIfMax()
+                var selectedCoinReferences = selectedCoinViewModels.Select(cvm => cvm.Model.OutPoint).ToList();
+
+                if (!selectedCoinReferences.Any())
+                {
+                    //SetWarningMessage("No coins are selected to spend.");
+                    return false;
+                }
+
                 var amount = Money.Zero;
-                Money.TryParse(AmountText.TrimStart('~', ' '), out amount);
-                return amount > 0;
+
+                var requests = new List<DestinationRequest>();
+
+                MoneyRequest moneyRequest;
+                if (IsMax)
+                {
+                    moneyRequest = MoneyRequest.CreateAllRemaining(subtractFee: true);
+                }
+                else
+                {
+                    if (!Money.TryParse(AmountText, out amount) || amount == Money.Zero)
+                    {
+                        // SetWarningMessage($"Invalid amount.");
+                        return false;
+                    }
+
+                    if (amount == selectedCoinViewModels.Sum(x => x.Amount))
+                    {
+                        // NotificationHelpers.Warning("Looks like you want to spend whole coins. Try Max button instead.", "");
+                        return false;
+                    }
+                    moneyRequest = MoneyRequest.Create(amount, subtractFee: false);
+                }
+
+                if (FeeRate is null || FeeRate.SatoshiPerByte < 1)
+                {
+                    return false;
+                }
+
+                var feeStrategy = FeeStrategy.CreateFromFeeRate(FeeRate);
+
+                var smartLabel = new SmartLabel(Label);
+                var activeDestinationRequest = new DestinationRequest(Address, moneyRequest, smartLabel);
+                requests.Add(activeDestinationRequest);
+                var intent = new PaymentIntent(requests);
+
+                var result = await Task.Run(() => Global.Wallet.BuildTransaction(
+                    password,
+                    intent,
+                    feeStrategy,
+                    allowUnconfirmed: true,
+                    allowedInputs: selectedCoinReferences));
+                SmartTransaction signedTransaction = result.Transaction;
+                SignedTransaction = signedTransaction;
+
+                await Global.TransactionBroadcaster.SendTransactionAsync(signedTransaction); // put this on non-ui theread?
+
+                return true;
             }
-            catch (Exception e)
+            catch (InsufficientBalanceException ex)
             {
-                return false;
+                Money needed = ex.Minimum - ex.Actual;
+                Logger.LogDebug(ex);
+                //SetWarningMessage($"Not enough coins selected. You need an estimated {needed.ToString(false, true)} BTC more to make this transaction.");
```

Comment:
Why did we not end up using `SetWarningMessage`?

Reply:
it's vestige of wasabi's notification manager. Probably better to add to the exception and then throw it to a toast in the view